Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -55,6 +56,76 @@ struct PackageController {
         }
     }
 
+    enum Fragment: String {
+        case css
+        case data
+        case documentation
+        case js
+        case themeSettings
+
+        var contentType: String {
+            switch self {
+                case .css:
+                    return "text/css"
+                case .data, .themeSettings:
+                    return "application/octet-stream"
+                case .documentation:
+                    return "text/html; charset=utf-8"
+                case .js:
+                    return "application/javascript"
+            }
+        }
+    }
+
+    func documentation(req: Request, fragment: Fragment) async throws -> Response {
+        guard
+            let owner = req.parameters.get("owner"),
+            let repository = req.parameters.get("repository"),
+            let reference = req.parameters.get("reference")
+        else {
+            throw Abort(.notFound)
+        }
+
+        let path = req.parameters.getCatchall().joined(separator: "/")
+
+        let url = try Self.awsDocumentationURL(owner: owner, repository: repository, reference: reference, fragment: fragment, path: path)
+        let res = try await Current.fetchDocumentation(req.client, url)
+        guard (200..<399).contains(res.status.code) else {
+            // Convert anything that isn't a 2xx or 3xx into a 404
+            throw Abort(.notFound)
```

Comment:
This is now showing our 404 instead of an AWS 404 (or other error page) with our docs header on top. That's better but eventually we'll probably want to redirect to a page that explains that there's a problem with doc generation like we do for build errors.